src/Fabulous.AST/Widgets/TypeDefinitions/Union.fs

```diff
@@ -34,20 +36,37 @@ module Union =
 
             let multipleAttributes =
                 match attributes with
-                | ValueSome values -> TypeHelpers.createAttributes values |> Some
+                | ValueSome values -> MultipleAttributeListNode.Create values |> Some
+                | ValueNone -> None
+
+            let typeParams = Helpers.tryGetScalarValue widget TypeParams
+
+            let typeParams =
+                match typeParams with
+                | ValueSome values ->
+                    TyparDeclsPostfixListNode(
+                        SingleTextNode.lessThan,
+                        [ for v in values do
+                              TyparDeclNode(None, SingleTextNode.Create v, Range.Zero) ],
+                        [],
+                        SingleTextNode.greaterThan,
+                        Range.Zero
+                    )
+                    |> TyparDecls.PostfixList
+                    |> Some
                 | ValueNone -> None
 
             TypeDefnUnionNode(
                 TypeNameNode(
                     None,
                     multipleAttributes,
-                    SingleTextNode("type", Range.Zero),
-                    Some(name),
-                    IdentListNode([ IdentifierOrDot.Ident(SingleTextNode("=", Range.Zero)) ], Range.Zero),
+                    SingleTextNode.``type``,
                     None,
+                    IdentListNode([ IdentifierOrDot.Ident(name) ], Range.Zero),
+                    typeParams,
                     [],
                     None,
-                    None,
+                    Some(SingleTextNode.equals),
```

Comment:
These parameters had to be moved around as they didn't create valid code with TypeParams otherwise.